pkg/volume/metrics_cached.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package volume
+
+import (
+	"sync"
+)
+
+var _ MetricsProvider = &metricsCached{}
+
+// metricsCached represents a MetricsProvider that wraps another provider and caches the result.
+type metricsCached struct {
```

Comment:
Oh, I see your just caching a single result... maybe `staticMetricsProvider`? or just leave it...